test/e2e/suite_test.go

```diff
@@ -51,78 +52,83 @@ var _ = BeforeSuite(func() {
+	// CSI driver is installed for that case.
+	if os.Getenv(driver.AzureDriverNameVar) == "" && testutil.IsRunningInProw() {
+		creds, err := credentials.CreateAzureCredentialFile(false)
+		Expect(err).NotTo(HaveOccurred())
+		azureClient, err := azure.GetAzureClient(creds.Cloud, creds.SubscriptionID, creds.AADClientID, creds.TenantID, creds.AADClientSecret)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = azureClient.EnsureResourceGroup(context.Background(), creds.ResourceGroup, creds.Location, nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Need to login to ACR using SP credential if we are running in Prow so we can push test images.
+		// If running locally, user should run 'docker login' before running E2E tests
 
-	// Need to login to ACR using SP credential if we are running in Prow so we can push test images.
-	// If running locally, user should run 'docker login' before running E2E tests
-	if testutil.IsRunningInProw() {
 		registry := os.Getenv("REGISTRY")
 		Expect(registry).NotTo(Equal(""))
 
 		log.Println("Attempting docker login with Azure service principal")
 		cmd := exec.Command("docker", "login", fmt.Sprintf("--username=%s", creds.AADClientID), fmt.Sprintf("--password=%s", creds.AADClientSecret), registry)
-		err := cmd.Run()
+		err = cmd.Run()
 		Expect(err).NotTo(HaveOccurred())
 		log.Println("docker login is successful")
-	}
 
-	// Install Azure Disk CSI Driver on cluster from project root
-	err = os.Chdir("../..")
-	Expect(err).NotTo(HaveOccurred())
-	defer func() {
-		err := os.Chdir("test/e2e")
+		// Install Azure Disk CSI Driver on cluster from project root
+		err = os.Chdir("../..")
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := os.Chdir("test/e2e")
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		projectRoot, err := os.Getwd()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(strings.HasSuffix(projectRoot, "azuredisk-csi-driver")).To(Equal(true))
+
+		log.Println("Installing Azure Disk CSI Driver...")
+		cmd = exec.Command("make", "e2e-bootstrap")
+		cmd.Dir = projectRoot
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
 		Expect(err).NotTo(HaveOccurred())
-	}()
-
-	projectRoot, err := os.Getwd()
-	Expect(err).NotTo(HaveOccurred())
-	Expect(strings.HasSuffix(projectRoot, "azuredisk-csi-driver")).To(Equal(true))
-
-	log.Println("Installing Azure Disk CSI Driver...")
-	cmd := exec.Command("make", "e2e-bootstrap")
-	cmd.Dir = projectRoot
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err = cmd.Run()
-	Expect(err).NotTo(HaveOccurred())
-	log.Println("Azure Disk CSI Driver installed")
-
-	nodeid := os.Getenv("nodeid")
-	azurediskDriver = azuredisk.NewDriver(nodeid)
-	go func() {
-		os.Setenv("AZURE_CREDENTIAL_FILE", credentials.TempAzureCredentialFilePath)
-		azurediskDriver.Run(fmt.Sprintf("unix:///tmp/csi-%s.sock", uuid.NewUUID().String()))
-	}()
+		log.Println("Azure Disk CSI Driver installed")
+
+		nodeid := os.Getenv("nodeid")
```

Comment:
check for empty string

Reply:
No changes done here...Oh...they appear in diff because of format change of adding if above. In any case will add the check. Thanks Rita.

Reply:
Done